libexec/bats-core/bats-preprocess

```diff
@@ -41,10 +41,13 @@ while IFS= read -r line; do
     name="${BASH_REMATCH[1]#[\'\"]}"
     name="${name%[\'\"]}"
     body="${BASH_REMATCH[2]}"
-    encode_name "$name" 'encoded_name'
-    tests["${#tests[@]}"]="$encoded_name"
+    bats_encode_test_name "$name" 'encoded_name'
     printf '%s() { bats_test_begin "%s" %d; %s\n' "$encoded_name" "$name" \
       "$index" "$body"
+
+    if [[ -z "$BATS_TEST_FILTER" || "$name" =~ $BATS_TEST_FILTER ]]; then
```

Comment:
I'm leaning against case insensitivity, but not strongly. Is it really significantly inconvenient for you, @sublimino? If it isn't, yet the case sensitivity reveals itself as a point of friction for users, perhaps we could add it later.

Reply:
It's a minor annoyance with the first time I tested this PR, that's all :) I'll feed back after I've used it in anger

Reply:
So are you good to go with this update? Feel free to merge if you like, or wait if you want @jasonkarns or others to chime in. (On the changes as they are, as well as whether I should throw in case insensitivity.)

I know I still owe @jasonkarns feedback on #124 as well... But after digging into this and queuing up some other cleanups and changes on top of this on Saturday, I gave myself a break on Sunday.